ProcessMaker/bpm
# PHP SDK for Docker

Depends on https://github.com/ProcessMaker/bpm/pull/1484
And https://github.com/ProcessMaker/executor-php/pull/2
Also the executor-php must be built and pushed to docker hub before this PR can be merged

Fixes #1472 
Fixes #1504 

Testing Info:
- Requires a branch from executor-php. So git clone that somewhere outside this repo
  `https://github.com/ProcessMaker/executor-php.git && cd executor-php && git checkout feature/1`
- composer install in the src folder `cd src && composer install`
- delete the old docker image if one exists `docker rmi processmaker/executor:php`
  - If it complains, stop any running containers then `docker container prune` and try the docker rmi command agian
- then back in the executor root, build the docker image `docker build -t processmaker/executor:php .`
- Now back in this repository, make sure you are running a NON-https bpm4 instance since the self-signed certificates have problems with the api
- Create a new php script with api calls
```
<?php
$emails = [];
foreach ($api->users()->getUsers()->getData() as $user) {
    $emails[] = $user->getEmail();
}
return ['emails' => $emails];
```
- Verify the script runs without errors and that you get the expected data. Add more users to see more emails in the example above.
- Reset db due to unrelated bug `php artisan migrate:fresh --seed --force && php artisan queue:restart`
- From the tests/Manual/PhpSDKScripts folder, import php_sdk_process.bpm4 and user_form_process.bpm4.
- Edit the scripts that got created from that import to set the run-as user to admin (we do not import users with the importer and all previous user associations are disassociated)
- Start a new 'PHP SDK Test' request
- (optional) Complete the new 'Pick color' task assigned to the logged in user
- Verify that after 5 minutes the 'PHP SDK Test' request completes automatically
